File: frontend/src/lib/api/themes.ts
import { useQuery } from "@tanstack/react-query";

import { apiClient } from "@/lib/api/client";

export interface ThemeTag {
  dimension_key: string;
  dimension_display_name: string;
  label_key: string;
  label_display_name: string;
  record_count: number;
}

export interface ThemeSummary {
  id: string;
  theme_key: string;
  name: string;
  theme_type: string;
  short_summary: string;
  representative_record_count: number;
  confidence_score: number | null;
  opportunity_score: number | null;
  tags: ThemeTag[];
}

export interface EvidencePreview {
  id: string;
  excerpt: string;
  source_connector_key: string;
  published_at: string | null;
}

export interface ThemeDetail extends ThemeSummary {
  long_summary: string | null;
  theme_set_id: string;
  representative_evidence: EvidencePreview[];
  counterexample: EvidencePreview | null;
}

export interface ThemeList {
  theme_set_id: string | null;
  theme_set_status: string | null;
  analysis_run_id: string | null;
  themes: ThemeSummary[];
}

export function useThemes() {
  return useQuery({
    queryKey: ["themes"],
    queryFn: () => apiClient.get<ThemeList>("/api/v1/themes"),
  });
}

export function useTheme(themeId: string | null) {
  return useQuery({
    queryKey: ["theme", themeId],
    queryFn: () => apiClient.get<ThemeDetail>(`/api/v1/themes/${themeId}`),
    enabled: Boolean(themeId),
  });
}

File: backend/src/instamart_engine/api/schemas/themes.py
"""Response DTOs for the (minimal, read-only) themes surface.
architecture.md §20.4 defines the full theme/insight CRUD surface; this is
just the read path the Phase 7 frontend needs — list + detail."""

from uuid import UUID

from pydantic import BaseModel


class ThemeTagResponse(BaseModel):
    dimension_key: str
    dimension_display_name: str
    label_key: str
    label_display_name: str
    record_count: int


class ThemeSummaryResponse(BaseModel):
    id: UUID
    theme_key: str
    name: str
    theme_type: str
    short_summary: str
    representative_record_count: int
    confidence_score: float | None
    opportunity_score: float | None
    tags: list[ThemeTagResponse]


class EvidencePreviewResponse(BaseModel):
    id: UUID
    excerpt: str
    source_connector_key: str
    published_at: str | None


class ThemeDetailResponse(ThemeSummaryResponse):
    long_summary: str | None
    theme_set_id: UUID
    representative_evidence: list[EvidencePreviewResponse]
    counterexample: EvidencePreviewResponse | None


class ThemeListResponse(BaseModel):
    theme_set_id: UUID | None
    theme_set_status: str | None
    analysis_run_id: UUID | None
    themes: list[ThemeSummaryResponse]
